src/utils/pgwire/src/pg_protocol.rs

```diff
@@ -199,37 +208,122 @@ where
         self.do_process(msg).await.is_none() || self.is_terminate
     }
 
+    /// The root tracing span for processing a message. The target of the span is
+    /// [`PGWIRE_ROOT_SPAN_TARGET`].
+    ///
+    /// This is used to provide context for the (slow) query logs and traces.
+    ///
+    /// The span is only effective if there's a current session and the message is
+    /// query-related. Otherwise, `Span::none()` is returned.
+    fn root_span_for_msg(&self, msg: &FeMessage) -> tracing::Span {
+        let Some(session_id) = self.session.as_ref().map(|s| s.id().0) else {
+            return tracing::Span::none();
+        };
```

Comment:
When is there no session? :eyes:

Reply:
I guess when the `Startup` message is not handled?